bot/exts/valentines/valentine_zodiac.py

```diff
@@ -19,36 +22,110 @@ class ValentineZodiac(commands.Cog):
+                await ctx.send(f"Sorry, but `{month}` is wrong month name.")
+                return
+        if (month == 1 and (1 <= query_date <= 19)) or (month == 12 and (22 <= query_date <= 31)):
+            zodiac = "capricorn"
+            final_embed = self.zodiac_sign_verify(zodiac)
+        else:
+            try:
+                zodiac_sign_based_on_month_and_date = self.zodiac_date_verifier(date(2020, month, query_date))
+                log.info("zodiac sign based on month and date received")
+            except ValueError as e:
+                log.info("invalid date or month given")
+                final_embed = discord.Embed()
+                final_embed.color = Colours.pink
+                final_embed.description = f"Zodiac sign is not found because, {e}"
+                log.info(e)
+            else:
+                final_embed = self.zodiac_sign_verify(zodiac_sign_based_on_month_and_date)
+                log.info("zodiac sign embed based on month and date is now sent.")
+
+        await ctx.send(embed=final_embed)
+
+    @zodiac.command(name="partnerzodiac")
+    async def partner_zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
         """Provides a counter compatible zodiac sign to the given user's zodiac sign."""
+        embed = discord.Embed()
+        embed.color = Colours.pink
         try:
             compatible_zodiac = random.choice(self.zodiacs[zodiac_sign.lower()])
+            emoji1 = random.choice(HEART_EMOJIS)
+            emoji2 = random.choice(HEART_EMOJIS)
+            embed.title = "Zodiac Compatibility"
+            embed.description = f"""{zodiac_sign.capitalize()}{emoji1}{compatible_zodiac["Zodiac"]}
+                            {emoji2}Compatibility meter : {compatible_zodiac["compatibility_score"]}{emoji2}"""
+            embed.add_field(
+                name=f'A letter from Dr.Zodiac {LETTER_EMOJI}',
+                value=compatible_zodiac['description']
+            )
         except KeyError:
-            return await ctx.send(zodiac_sign.capitalize() + " zodiac sign does not exist.")
-
-        emoji1 = random.choice(HEART_EMOJIS)
-        emoji2 = random.choice(HEART_EMOJIS)
-        embed = discord.Embed(
-            title="Zodic Compatibility",
-            description=f'{zodiac_sign.capitalize()}{emoji1}{compatible_zodiac["Zodiac"]}\n'
-                        f'{emoji2}Compatibility meter : {compatible_zodiac["compatibility_score"]}{emoji2}',
-            color=Colours.pink
-        )
-        embed.add_field(
-            name=f'A letter from Dr.Zodiac {LETTER_EMOJI}',
-            value=compatible_zodiac['description']
-        )
+            err_comp = [f"`{i}` {zod_name}" for i, zod_name in enumerate(self.zodiac_fact.keys(), start=1)]
+            error = ("\n").join(err_comp)
+            error_msg = f"**{zodiac_sign}** is not a valid zodiac sign, here is the list of valid zodiac signs."
+            embed.description = f"{error_msg}\n{error}"
```

Comment:
This part of the code is being repeated. It's better to make a function for it.
repeated at line 56